src/common/entity/compute_state_display.ts

```diff
@@ -29,37 +29,77 @@ export const computeStateDisplay = (
   const domain = computeStateDomain(stateObj);
 
   if (domain === "input_datetime") {
-    let date: Date;
-    if (!stateObj.attributes.has_time) {
+    if (state) {
+      // If trying to display an explicit state, need to parse the explict state to `Date` then format.
+      // Attributes aren't available, we have to use `state`.
+      try {
+        // We are going to convert the state string into a ISO8601 string,
+        // which is always in UTC and causing the tiemzone info to be lost,
+        // therefore we need to compensate the local timezone offset.
+        const timezoneOffset = new Date().getTimezoneOffset() * 60 * 1000;
+        const components = state.split(" ");
+        if (components.length === 2) {
+          // Date and time.
+          const iso8601String = components.join("T") + "Z";
```

Comment:
If you leave the `Z` of, you don't need the `timezoneOffset`

Reply:
I thought `Z` is required in ISO8601 spec, even after reading some documents, it's still vague.

But I verified it works without `Z` on both Chrome and Safari, I will remove it. Thanks for the reminder!